torch/_inductor/scheduler.py

```diff
@@ -405,6 +416,59 @@ def can_inplace(self, read_dep: dependencies.MemoryDep):
             return read_dep.index == write_dep.index and read_dep.size == write_dep.size
         return False
 
+    def reorder_loop(
+        self, reorder: Callable[[sympy.Expr], sympy.Expr], ndim: int
+    ) -> Optional["SchedulerNode"]:
+        """Returns a new node which represents the same loop iterated in a new order
+
+        reorder is a function that takes indices in the new order and
+        translates them into the indices expected by the underlying loop. e.g.
+
+            reorder = lambda index: index[1], index[0]
+
+        will swap the order of a 2d loop. Note this only applies to the
+        pointwise index variables.
+
+        Also note some class members are shallow copied to the new node, so do
+        not modify the returned node.
+
+        """
+
+        if len(self.get_ranges()[0]) != ndim:
```

Comment:
Should this be an error?  When does this happen?

Reply:
This is mainly for the benefit of `FusedSchedulerNode.reorder_loop` since cuda can fuse nodes that have a common numel, but not a common shape. I could move the line there if you prefer.